lib/commandCenter/commandCenter.js

```diff
@@ -388,15 +387,53 @@ function compress(content, cb) {
 	});
 }
 
+/**
+ * Promisify the user command execute method, or
+ * ensure its return value is passed to state.respond
+ *
+ * Once callback-based user commands are removed from MAGE,
+ * you will be able to remove this method.
+ *
+ * @param {*} cmdInfo
+ * @param {State} state
+ * @param {*} paramList
+ */
+function generateCommandPromise(cmdInfo, state, paramList) {
+	const mod = cmdInfo.mod;
+	const execute = mod.execute.apply.bind(mod.execute, mod, paramList);
+
+	if (cmdInfo.isAsync) {
+		// Async user commands will need to take the response value, and feed it
+		// to `state.respond`
+		return execute()
+			.then((response) => state.respond(response, mod.serialize === false));
+	} else {
+		// Callback-based user commands are expected to call `state.respond`
+		// on their own; all we need to do is promisify
+		return new Promise((resolve) => mod.execute(resolve));
```

Comment:
Should this not say `execute` instead of `mod.execute`?

Reply:
If `execute` uses `this` then calling just `execute` would break unexpectedly I believe.

Reply:
Whoops. nope. You are right.